pinot-core/src/main/java/org/apache/pinot/core/minion/SegmentPurger.java

```diff
@@ -165,6 +160,7 @@ public boolean hasNext() {
 
         // Try to get the next row to return
         while (_pinotSegmentRecordReader.hasNext()) {
+          _nextRow.clear();
```

Comment:
Should the clear be inside of next()? So that callers don't need to explicit call?

Reply:
Yes, but that requires changing all the record readers. I added a TODO in this PR, and we can address this in a separate PR.